hcloud/client_handler_retry.go

```diff
@@ -30,13 +32,54 @@ func (h *retryHandler) Do(req *http.Request, v any) (resp *Response, err error)
 			// - request preparation
 			// - network connectivity
 			// - http status code (see [errorHandler])
-			if IsError(err, ErrorCodeConflict) {
-				time.Sleep(h.backoffFunc(retries))
-				retries++
-				continue
+			if ctx.Err() != nil {
+				return resp, ctx.Err()
+			}
```

Comment:
I think we should mention in the release notes that we now respect cancelled contexts.